configtemplate/metadata/pivnet.go

```diff
@@ -177,28 +181,24 @@ func productFileKeysByGlobs(
 
 type AuthenticatedPivnetClient struct {
 	ClientConfig pivnetapi.ClientConfig
+	TokenService pivnetapi.AccessTokenService
 	HTTPClient   *http.Client
 }
 
 func (a AuthenticatedPivnetClient) Do(req *http.Request) (*http.Response, error) {
-	const legacyAPITokenLength = 20
-	if len(a.ClientConfig.Token) > legacyAPITokenLength {
```

Comment:
All this logic is now taken care of by the TokenService and the `AuthorizationHeader` function.